src/core/RemotePlugin.cpp

```diff
@@ -404,32 +403,31 @@ bool RemotePlugin::process( const SampleFrame* _in_buf, SampleFrame* _out_buf )
 	waitForMessage( IdProcessingDone );
 	unlock();
 
-	const ch_cnt_t outputs = std::min<ch_cnt_t>(m_outputCount,
-							DEFAULT_CHANNELS);
-	if( m_splitChannels )
+	const ch_cnt_t outputsClamped = std::min<ch_cnt_t>(m_pinConnector.out().channelCount, DEFAULT_CHANNELS);
+
+	if (m_splitChannels)
 	{
-		for( ch_cnt_t ch = 0; ch < outputs; ++ch )
-		{
-			for( fpp_t frame = 0; frame < frames; ++frame )
-			{
-				_out_buf[frame][ch] = m_audioBuffer[( m_inputCount+ch )*
-								frames + frame];
-			}
-		}
+		// NOTE: VST plugins always use split channels
+		const auto trackChannels = CoreAudioDataMut{&_out_buf, 1};
+		const auto pluginOutput = SplitAudioData<const sample_t> {
+			m_audioBuffer.get() + (frames * m_pinConnector.in().channelCount),
+			static_cast<std::size_t>(frames * m_pinConnector.out().channelCount)
+		};
+
+		m_pinConnector.routeFromPlugin(frames, pluginOutput, trackChannels);
 	}
-	else if( outputs == DEFAULT_CHANNELS )
+	else if (outputsClamped == DEFAULT_CHANNELS)
 	{
-		auto source = m_audioBuffer.get() + m_inputCount * frames;
+		auto source = m_audioBuffer.get() + m_pinConnector.in().channelCount * frames;
 		copyToSampleFrames(_out_buf, source, frames);
```

Comment:
I think I'd rather keep it as an alias that just serves as a kind of annotation without changing the actual type (i.e. not a strong typedef). It won't prevent misuse like this, but I think it's better than plain `float*`.

In the future we may want to consider converting all "Interleaved" audio data in LMMS to "Split". Lv2, LADSPA, CLAP, and VST all use "Split" audio data, but LMMS internally and LMMS plugins use "Interleaved". I'm not sure about audio devices or 3rd party libraries though.

Reply:
I'm reconsidering whether to make it a strong typedef using a technique like Boost's `BOOST_STRONG_TYPEDEF`.
It would allow me to overload the `routeToPlugin` and `routeToPlugin` methods for interleaved and split layouts and also prevent misuse. My one concern is whether it can be done in a way where the compiler will know it is trivially copyable and not cause any performance penalty.